microsoft/vscode
# Add support for sparse-checkout mode in Git add command to resolve is…

This pull request addresses issue #213677 by adding support for sparse-checkout mode in VSCode's Git integration. The change ensures that when sparse-checkout is enabled, the git add command includes the --sparse flag, allowing files outside the sparse-checkout definition to be properly staged and committed.

How to test?
```git clone --no-checkout https://myproject.git``` or init a new repo;
```git sparse-checkout init```;
create a new file;
try to stage file with add button on resource state from git extension in source control.

Result: The file should be added to index with no problem!
